proposals/scopes.md

```diff
@@ -112,66 +112,136 @@ The following information describes a scope in the source map:
   - the original variable name
   - a javascript expression that can be evaluated by the debugger in the corresponding generated scope to get the binding's value (if such an expression is available)
 
+The following code snippet specifies the scope information **conceptually** in TypeScript notation. See the [Encoding](#encoding) section
+on how this information is actually VLQ encoded.
+
+```ts
+interface SourceMap {
+  // ...
+  originalScopes?: OriginalScope[];
+  generatedScopes?: GeneratedScope;
+}
+
+interface OriginalScope {
+  start: OriginalPosition;
+  end: OriginalPosition;
+  kind: ScopeKind;
+  /** Class/module/function name. Can be used for stack traces or naming scopes in a debugger's scope view */
+  name?: string;
+  /** Symbols defined in this scope and where they are found in the generated code (plus how to get their values). */
+  bindings: Map<string, BindingRange[]>;
+  children?: OriginalScope[];
+}
+
+interface GeneratedScope {
+  start: GeneratedPosition;
+  end: GeneratedPosition;
+  originalScope?: OriginalScope;
+  /** If this scope corresponds to an inlined function body, record the callsite of the inlined function in the original code */
+  callsite?: OriginalPosition;
+  children?: GeneratedScope[];
+}
+
+type ScopeKind = 'global' | 'class' | 'function' | 'block';
+
+interface BindingRange {
+  from: GeneratedPosition;
+  to: GeneratedPosition;
```

Comment:
I don't see any definition for what the `from` and `to` in the BindingRange is intended to encompass, clarification may be useful